test/e2e/common/util_cleanup.go

```diff
@@ -6,95 +6,136 @@ import (
 	"os/exec"
 	"time"
 
-	"k8s.io/apimachinery/pkg/runtime/schema"
-
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+var ZeroInt64 = int64(0)
+
 /// Delete all pods in the default namespace
-// returns:
-// 1) success i.e. true if all pods were deleted or there were no pods to delete.
-// 2) the number of pods found
-func DeleteAllPods() (bool, int) {
+func DeleteAllPods() (int, error) {
 	logf.Log.Info("DeleteAllPods")
-	success := true
 	numPods := 0
+
 	pods, err := gTestEnv.KubeInt.CoreV1().Pods("default").List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		logf.Log.Info("DeleteAllPods: list pods failed.", "error", err)
-		success = false
-	}
-	if err == nil && pods != nil {
+	} else {
 		numPods = len(pods.Items)
+		logf.Log.Info("DeleteAllPods: found", "pods", numPods)
 		for _, pod := range pods.Items {
 			logf.Log.Info("DeleteAllPods: Deleting", "pod", pod.Name)
-			if err := DeletePod(pod.Name); err != nil {
-				success = false
+			delErr := gTestEnv.KubeInt.CoreV1().Pods("default").Delete(context.TODO(), pod.Name, metav1.DeleteOptions{GracePeriodSeconds: &ZeroInt64})
+			if delErr != nil {
+				logf.Log.Info("DeleteAllPods: failed", "pod", pod.Name, "error", delErr)
```

Comment:
Should this be an Error rather than an Info?
Also, the message `DeleteAllPods: failed` seems to suggest we failed to delete all pods rather than a specific instance.

Reply:
I am conflicted a bit on whether to user Error, AFAICT using Error gives us a backtraces, which are only useful for failures intrinsic to the test as opposed to entities being tested. Which includes the cluster IMO.

Reply:
I've amended the log message to `"DeleteAllPods: failed to delete a pod", "podName"...`